openraft/src/storage.rs

```diff
@@ -111,6 +111,13 @@ where
     /// for details on where and how this is used.
     type SnapshotData: AsyncRead + AsyncWrite + AsyncSeek + Send + Sync + Unpin + 'static;
 
+    /// Initialize the storage(e.g. state).
+    ///
+    /// When the Raft node is first started, it will call this interface on the storage system to
+    /// initialize node related information,e.g. node state then `InitialState::new_initial` should
+    /// be used.
+    async fn initialize(&self) -> Result<(), StorageError>;
```

Comment:
IMO, a store does not need to provide an API  `initialize()`. 🤔 

When initializing, RaftCore could just do this:
```
let state = self.sto.get_initial_state().await?;
if state.is_none() {
        let state = InitialState::new_initial(self.id);
        self.sto.save_hard_state(&state.hard_state).await?;
}
```

Reply:
okk, revise immediately